mos_tests/murano/test_kubernetes.py

```diff
@@ -263,3 +265,96 @@ def test_kub_nodes_down_if_one_present(murano, environment, session, cluster,
     logs = murano.get_log(deployed_environment)
     assert 'Action scaleGatewaysDown is scheduled' in logs
     assert 'No gateway nodes that can be removed' in logs
+
+
+@pytest.mark.check_env_("is_any_compute_suitable_for_max_flavor")
+@pytest.mark.parametrize('pod', '2', indirect=['pod'])
+@pytest.mark.testrail_id('836661')
+def test_pod_replication(env, os_conn, keypair, murano, environment, session,
+                         cluster, pod, dockerhttpd):
+    """Check that replication controller works correctly
+    Scenario:
+        1. Create Murano environment
+        2. Add Kubernetes Cluster application to the environment: Set
+        initial_gateways=1, max_gateways=1, initial_nodes=1, max_nodes=1
+        3. Add Kubernetes Pod to the environment
+        4. Add some application to the environment
+        5. Deploy environment
+        6. Check deployment status and make sure that all nodes are active
+        7. Go to minion node and check count of pod's replicas
+        8. Kill one replica
+        9. Wait for new pod's replica creation. Check that id of newly created
+        replica differs from id of removed replica
+        10. Remove environment
+    """
+    deployed_environment = murano.deploy_environment(environment, session)
+    murano.check_instances(gateways_count=1, nodes_count=1)
+    murano.status_check(deployed_environment,
+                        [[cluster['name'], "master-1", 8080],
+                         [cluster['name'], "gateway-1", 80],
+                         [cluster['name'], "minion-1", 4194]], kubernetes=True)
+    srv = [i for i in os_conn.nova.servers.list() if 'minion' in i.name][0]
+    with os_conn.ssh_to_instance(env, srv, vm_keypair=keypair,
+                                 username='ubuntu') as remote:
+        def get_pods():
+            return remote.check_call('sudo docker ps | grep httpd')['stdout']
+        res = get_pods()
+        assert len(res) == 2, "{0} replicas instead of {1}".format(len(res), 2)
+        pod_id = [p.split(' ')[0] for p in res][0]
+        remote.check_call('sudo docker kill {0}'.format(pod_id))
+        wait(lambda: len(get_pods()) == 2, timeout_seconds=120,
+             waiting_for="pod to be recreated")
+        new_ids = [p.split(' ')[0] for p in get_pods()]
+        assert new_ids.count(pod_id) == 0, \
+            "No new pod's replica added, {0} still in the list".format(pod_id)
+
+
+@pytest.mark.check_env_("is_any_compute_suitable_for_max_flavor")
+@pytest.mark.parametrize('pod', '2', indirect=['pod'])
+@pytest.mark.testrail_id('836663', params={'action': 'Up'})
+@pytest.mark.testrail_id('836664', params={'action': 'Down'})
+@pytest.mark.parametrize('action', ['Up', 'Down'])
```

Comment:
Let's add one more parameter - like `expected_count`
It will looks like
```python
@pytest.mark.parametrize('action, expected_count', [('Up', 3), ('Down', 1)])
def test_pod_action_up_down(action, expected_count, ...):
```
And next you can make single compare - real pod count with expected